general/session.md

```diff
@@ -0,0 +1,26 @@
+# Session API
+
+## Workflow
+
+### Getting
+
+When getting a value stored in Session, we are deserializing it by removing the `serialize://` string from the data we retrieve. After the data is deserialized we get access to the data requested, stripped of any serialization.
+
+#### Needs clarifying
+
+-   Even though we are using POST we are sending a `'cmd': 'get'`, which might be revealing the underlying architecture of how Liferay handles fetch requests.
+-   Why are we using a POST method in a getter?
+
+### Setting
+
+When setting values to Session, we have to serialize them first so that it is served in a proper way that the endpoint expects. After the value has been serialized, we append it to FormData which then get's sent to the `session_click` endpoint to be consumed at a later time.
```

Comment:
Doesn't seem to describe what is actually happening. Maybe something like?

---

`setSessionValue(key, data)` makes a request to the server to associate `data` with `key` in the user's session. If `data` is a non-falsy object, it will be encoded with `JSON.stringify()` and prefixed with a special marker, `serialize://`, to flag it for deserialization on later retrieval. Otherwise, it gets transmitted as-is.